File: python_files/embryo_dev_stage_solve.py
# enter temperature here:
temp = 23

# enter hours here:
time = 53.5

import numpy as np
from scipy.optimize import root_scalar

f1 = lambda nfstage: -137.4 * np.exp(-0.1460 * temp) + (21.54 * np.exp(-0.1192 * temp)) * nfstage - time

stage = round((root_scalar(f1, bracket=[0, 120], method='brentq')).root, 1)

if stage <= 40:
    print(stage)
else:
    f1 = lambda nfstage: -2481.0 * np.exp(-0.09332 * temp) + (81.97 * np.exp(-0.09745 * temp)) * nfstage - time

    stage = round((root_scalar(f1, bracket=[0, 50], method='brentq')).root, 1)
    print(f"NF Stage {stage}")

File: embryo_dev_stage_solve.R


#enter temperature here:
temp<-23

#enter hours here:
time<- 53.5



library("rootSolve")

f1 = function(nfstage){
    -137.4*exp(-0.1460*temp) + (21.54*exp(-0.1192*temp))*nfstage-time
  }

stage = round((uniroot.all(f1, c(0,120))), digits = 1)

if(stage<=40){
  print(stage)
  
} else {
  f1 = function(nfstage){
    -2481.0*exp(-0.09332*temp) + (81.97*exp(-0.09745*temp))*nfstage-time
  }
  
  stage = round((uniroot.all(f1, c(0,50))), digits = 1)
  print(stage)
}





